### PR TITLE
do not crash invoice if wellknown metadata keys used with different e…

### PR DESCRIPTION
**Pull request by Kukks**
…xpected types

### CODE REVIEW

**Comment by NicolasDorier** [comment]
I don't really like we try to be too smart here. Let me give a try.

File: BTCPayServer/Services/Invoices/InvoiceEntity.cs [hunk 1]
```diff
@@ -30,70 +30,136 @@ static InvoiceMetadata()
...
+            }
+            else
+            {
+                try
                 {
-                    try
+                    if (value is string s)
                     {
-                        PosRawData = JToken.Parse(value);
+                        data = JToken.Parse(s);
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Invoices/InvoiceEntity.cs:146]: this assume s is representing json, this is not the case. A String of `\"lol\"` should not be interpreted as json `"blah" :"lol"`

**Comment by NicolasDorier** [comment]
@Kukks can you also add some stuff I did on the other PR which fix crash if there is duplicate field in the JSON in the posdata_parser (just replace .Add by TryAdd)

File: BTCPayServer/Services/Invoices/InvoiceEntity.cs (hunk 1 above)
Comment (Kukks) [reply BTCPayServer/Services/Invoices/InvoiceEntity.cs:146]: No, it actually works this way.

for example, creating an invoice with metadata:
```
    "metadata": {
            "taxIncluded" : 1,
            "buyerName" : "\"hello\"",
        "posData" : {
            "test": [ "hello", "bye"]
        }
    }
```
results in following: 
image

**Comment by Kukks** [comment]
> @Kukks can you also add some stuff I did on the other PR which fix crash if there is duplicate field in the JSON in the posdata_parser (just replace .Add by TryAdd)

Done

**Pull request merged**